src/language/lexer.js

```diff
@@ -284,6 +285,38 @@ function unexpectedCharacterMessage(code: number): string {
   return `Cannot parse the unexpected character ${printCharCode(code)}.`;
 }
 
+/**
+ * Reads a dot token with helpful messages for negative lookahead.
+ *
+ * \.(?![\.0-9])
+ */
+function readDot(
+  source: Source,
+  start: number,
+  line: number,
+  col: number,
+  prev: Token | null,
+) {
+  const nextCode = source.body.charCodeAt(start + 1);
+  if (nextCode === 46) {
+    // .
+    throw syntaxError(
+      source,
+      start,
+      'Cannot parse the unexpected character "..".',
+    );
+  }
+  if (nextCode >= 48 && nextCode <= 57) {
+    // 0 - 9
+    throw syntaxError(
+      source,
+      start,
+      'Invalid number, expected digit before ".".',
```

Comment:
maybe i'm just understanding this wrong, but it looks the the issue that we found a number _after_ "."?

Would `Unexpected number after "."` be more appropriate?

Reply:
You're exactly right - this error is from `. [lookahead != Digit]`, so a number coming after the "." - however the message that we provide for the failure should be as informative as possible based on the intent.

The rationale for limiting this is that in many languages, including javascript, `.123` is actually a legal number token, but in GraphQL it is not, and the leading `0.` is required. Many new to GraphQL may write `.123` directly in a query, expecting it to be interpreted as a number and ideally we can give a clear message as to why.

If you wrote a query: `{ doThing(num: .123) }` and get back `'Unexpected number after "."'` you might be confused given the intent - you *meant* to put a number after the `.` it wasn't a typo. Instead we should make it clear that this style of number isn't supported and how to fix it.

I wonder if we should make this message even more specific about needing a `0` before the `.`?